### PR TITLE
Modify the log of kube-proxy

### PR DESCRIPTION
**Pull request by viviyww**
Modify the warning log of kube-proxy when we run kube-proxy server
with --proxy-mode, but in the config file, we omit it, like 'kube-proxy --config config.conf  --proxy-mode ipvs'
Then it logs like ""{"log":"W0905 09:14:40.321571       1 server_others.go:249] Flag proxy-mode=\"\" unknown, assuming iptables proxy\n","stream":"stderr", "time":"2019-09-05T09:14:40.321858964Z"} 
This may lead to confusion. I think it should me modefied.

### CODE REVIEW

File: cmd/kube-proxy/app/server_others.go [hunk 1]
```diff
@@ -326,7 +326,7 @@ func getProxyMode(proxyMode string, khandle ipvs.KernelHandler, ipsetver ipvs.IP
 	case proxyModeIPVS:
 		return tryIPVSProxy(khandle, ipsetver, kcompat)
 	}
-	klog.Warningf("Flag proxy-mode=%q unknown, assuming iptables proxy", proxyMode)
+	klog.Warningf("Value of proxy mode: %q unknown, it may be overwritten by config file if flag --config is added, assuming iptables proxy", proxyMode)
```
Comment (neolit123) [inline cmd/kube-proxy/app/server_others.go]: does the config value really take precedence over the flag value?

Comment (neolit123) [reply cmd/kube-proxy/app/server_others.go]: i would change this to:
`Unknown proxy mode %q, assuming iptables proxy`

users would still know where to set the mode.

Comment (viviyww) [reply cmd/kube-proxy/app/server_others.go]: The config value will take precedence over the flag.

Comment (neolit123) [reply cmd/kube-proxy/app/server_others.go]: yeah, this is counter intuitive.
i would still recommend changing this line to `Unknown proxy mode %q, assuming iptables proxy`

Comment (viviyww) [reply cmd/kube-proxy/app/server_others.go]: Ok, I will recheck it. Thanks for review it.

**Comment by danwinship** [comment]
It seems to me like you are just trading one confusing log message for another? If the user has validly specified a proxy mode, then we should not log any error.

**Comment by viviyww** [comment]
> 
> 
> It seems to me like you are just trading one confusing log message for another? If the user has validly specified a proxy mode, then we should not log any error.

Yes, but the flag will be taken precedence over the config file.

**Comment by danwinship** [comment]
> Yes, but the flag will be taken precedence over the config file.

No it won't. If you specify `--config`, then the values specified there take precedence over almost all other command-line options. That's the problem here. You are incorrectly trying to specify the proxy mode on the command line while also using a config file, and that's not allowed. The error message is correctly informing you that since the config file didn't specify a proxy mode, that it is using the default mode.

**Comment by viviyww** [comment]
> 
> 
> > Yes, but the flag will be taken precedence over the config file.
> 
> No it won't. If you specify `--config`, then the values specified there take precedence over almost all other command-line options. That's the problem here. You are incorrectly trying to specify the proxy mode on the command line while also using a config file, and that's not allowed. The error message is correctly informing you that since the config file didn't specify a proxy mode, that it is using the default mode.

Thanks for your review. I agree with your idea and I add another issue to add mixed argement validation. As you says, --config and --proxy-mode together should not be allowed. But the log info here is not exatly right. So, in this issue, I just modify the log info.

**Pull request merged**